lib/eventFactory.js

```diff
@@ -519,6 +521,10 @@ class EventFactory extends BaseFactory {
         let prevChainPR = '';
         let decoratedCommit;
 
+        if (groupEventId) {
```

Comment:
we gonna need to add a logic to set groupEventId to current event's id if groupEventId was not passed in

Reply:
I think it's okay to leave `groupEventId` blank if it should be its own `eventId`.
Unless you think it makes more sense to try to update the event immediately after creation everywhere.

Reply:
Other logic for setting groupEventId is done in screwdriver repo: https://github.com/screwdriver-cd/screwdriver/pull/2010

Reply:
main reason I am bringing this up is that we should be able to query all event in same group by just using groupEventId, otherwise we would need to query second time to fetch root event.

Reply:
yeah that makes sense

If we want to update the newly created event with `groupEventId`, I don't think it can be done in the eventFactory model
we would need to duplicate that logic anytime we called `eventFactory.create()`

Reply:
since we have newly created `event` here https://github.com/screwdriver-cd/models/blob/9a3313761c94a627b500ac4c71ea6fbab1ff0eb7/lib/eventFactory.js#L637

can we add `groupEventId`  here if it doesn't exist and call `updated()`

Reply:
I tried that but I get 
```
TypeError: this[datastore].update is not a function
```

Reply:
do `event.update()`, this in eventFactory so we can't do `this.update()`

Reply:
updated